MetaMask/metamask-extension
# fix: error during persisted transaction approval

## Explanation
This pull request addresses the issue that occurs when a pending transaction is present, the extension is disabled and then re-enabled, and subsequently, the user rejects the transaction. The goal is to prevent the logging of the error message "User rejected the request", while still capturing and logging any other types of errors.

Fixes https://github.com/MetaMask/metamask-extension/issues/20604

## Screenshots/Screencaps




### After

[fix-error-during-persisted-transaction-approval.webm](https://github.com/MetaMask/metamask-extension/assets/45455812/518c844a-4ae0-443c-a0bc-34591e3bc43f)





## Manual Testing Steps
- Trigger any transaction
- Disable the wallet from chrome://extensions/ (switch button)
- Enable it again
- Unlock MM
- Reject the transaction
- Check console



## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
